Repositories/NoteRepository.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Taskmanager.Data.Context;
+using Taskmanager.Data.Entities;
+using Taskmanager.Repositories.Interfaces;
+
+namespace Taskmanager.Repositories
+{
+    public class NoteRepository : INoteRepository
+    {
+        private readonly TaskManagerContext _context;
+
+        public NoteRepository(TaskManagerContext context)
+        {
+            _context = context;
+        }
+
+        public async Task AddAsync(int userId, int todolistId, int todoitemId, Note note)
+        {
+            Todolist todolist = await _context.Todolists.FirstOrDefaultAsync(list => list.Userid == userId && list.Id == todolistId);
+
+            await _context.Notes.AddAsync(note);
```

Comment:
No need to call AddAsync, it's used for specific id generators. https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.addasync?view=efcore-5.0